samples/tutorial/Tutorial5.md

```diff
@@ -0,0 +1,612 @@
+            workflowType = TodoEditWorkflow::class,
+            rendering = TodoEditScreen(
+                title = "Title",
+                note = "Note",
+                onTitleChanged = {},
+                onNoteChanged = {},
+                discardChanges = {},
+                saveChanges = {}
+            ),
+            // Simulate it emitting an output of `.save` to update the state.
+            output = WorkflowOutput(
+                Save(
+                    TodoModel(
+                        title = "Updated Title",
+                        note = "Updated Note"
+                    )
+                )
+            )
+        )
+        .render { rendering ->
+          // Just validate that there are two items in the back stack.
+          // Additional validation could be done on the screens returned, if desired.
+          assertEquals(2, rendering.size)
+        }
+        // Validate that the state was updated after the render pass with the output from the
+        // TodoEditWorkflow.
+        .verifyActionResult { newState, _ ->
+          assertEquals(
+              State(
+                  todos = listOf(TodoModel(title = "Updated Title", note = "Updated Note")),
+                  step = List
+              ),
+              newState
+          )
+        }
+  }
+}
+```
+
+## Integration Testing
+
+The `RenderTester` allows easy "mocking" of child workflows and workers. However, this means that we are not exercising the full infrastructure (even though we could get a fairly high confidence from the tests). Sometimes, it may be worth putting together integration tests that test a full tree of Workflows. This lets us test integration with the non-workflow world as well, such as external reactive data sources that your workflows might be observing via Workers.
+
+Add another test to `RootWorkflowTests`. We will use another test helper that spins up a real instance of the workflow runtime, the same runtime that `setContentWorkflow` uses.
+
+### WorkflowTester
+
+When you create an Android app using Workflow, you will probably use `setContentWorkflow`, which starts a runtime to host your workflows, and wires it up to all the Android infrastructure. Under the hood, there's a lower-level API called `renderWorkflowIn` that runs the workflow runtime in a coroutine and exposes a `StateFlow` of renderings. When writing integration tests for workflows, you can use this API directly (maybe with a library like [Turbine](https://github.com/cashapp/turbine)), or you can use `workflow-testing`'s `WorkflowTester`. The `WorkflowTester` starts a workflow and lets you request renderings and outputs manually so you can write tests that interact with the runtime from the outside.
+
+This will be a "black box" test, as we can only test the behaviors from the rendering and will not be able to inspect the underlying states. This may be a useful test for validation when refactoring a tree of workflows to ensure they behave the same way.
```

Comment:
```suggestion
This will be an opaque test, as we can only test the behaviors from the rendering and will not be able to inspect the underlying states. This may be a useful test for validation when refactoring a tree of workflows to ensure they behave the same way.
```